lib/workflow/stores/mongo.js

```diff
@@ -677,11 +695,22 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _) {
 
         var query = {
             graphId: data.graphId,
-            reachable: true
+            reachable: true,
+            $or:[]
         };
-        query['dependencies.' + data.taskId] = {
+ 
+        var dependenciesItr = {};
+        dependenciesItr['dependencies.' + data.taskId] = {
             $in: _.difference(Constants.Task.FinishedStates, [data.state])
         };
+        query.$or.push(dependenciesItr);
+
+        delete dependenciesItr['dependencies.' + data.taskId];
+        dependenciesItr['dependencies.anyOf.' + data.taskId] = {
+            $in: _.difference(Constants.Task.FinishedStates, [data.state])
+        };
```

Comment:
For `anyOf`, only if all criterias are never been meet, then it can be though of unreachable.